debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -182,6 +186,13 @@ public boolean executeIteration(ChangeEventSourceContext context, SqlServerParti
                             schemaChangeCheckpoints.add(table);
                         }
                     }
+
+                    for (SqlServerChangeTable table : tables) {
+                        if (table.getStopLsn().isAvailable()) {
+                            LOGGER.info("The stop lsn of {} change table became known", table);
+                            changeTablesWithKnownStopLsn.add(table);
```

Comment:
Is the list always expected to be empty when this code is executed? Otherwise, what will happen if the same table is added twice? Maybe it should be a `Set` instead of a `List`? If duplicates are not expected, maybe we should enforce this here somehow? I.e. expect that `Set#add()` always returns `true`.

Reply:
Jira ticket: refactor this to use a set instead of a list.